core/modules/qdisp/QueryRequest.cc

```diff
@@ -151,24 +124,21 @@ bool QueryRequest::ProcessResponse(XrdSsiRespInfo const& rInfo, bool isOk) {
 
 /// Retrieve and process results in using the XrdSsi stream mechanism
 bool QueryRequest::_importStream() {
-    bool retrieveInitiated = false;
-    // Pass ResponseRequester's buffer directly.
-    std::vector<char>& buffer = _requester->nextBuffer();
+    bool success = false;
+    // Pass ResponseHandler's buffer directly.
+    std::vector<char>& buffer = _jobDesc.respHandler()->nextBuffer();
     LOGF_DEBUG("QueryRequest::_importStream buffer.size=%1%" % buffer.size());
     const void* pbuf = (void*)(&buffer[0]);
     LOGF_INFO("_importStream->GetResponseData size=%1% %2% %3%" %
```

Comment:
Yes! We are making this mistake in so many places I stopped pointing it out. I am hoping one day we will do a complete sweep through the code and move info to debug.